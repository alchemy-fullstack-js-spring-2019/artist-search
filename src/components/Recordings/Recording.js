import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Recording({ title, artist }) {
  return (
    <p>
      <Link to={ `/${artist}/recording/${title}` }>{ title }</Link>
    </p>
  );
}

Recording.propTypes = {
  title: PropTypes.string.isRequired,
  artist: PropTypes.string.isRequired
};

export default Recording;
