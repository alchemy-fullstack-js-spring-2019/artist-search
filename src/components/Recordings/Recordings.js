import React from 'react';
import PropTypes from 'prop-types';
import Recording from './Recording';

export default function Recordings({ artist, recordingArray }) {
  const recordingList = recordingArray.map(recording => {
    return (
      <li key={ recording.id }>
        <Recording artist={ artist } title={ recording.title } />
      </li>
    );
  });
  
  return (
    <ul>
      { recordingList }
    </ul>
  );
}

Recordings.propTypes = {
  artist: PropTypes.string.isRequired,
  recordingArray: PropTypes.array.isRequired
};
