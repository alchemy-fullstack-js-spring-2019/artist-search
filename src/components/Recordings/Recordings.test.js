import React from 'react';
import { shallow } from 'enzyme';
import Recordings from './Recordings';

describe('Recordings component', () => {
  it('renders the Recordings', () => {
    const fakeData = [
      {
        title: 'Stronger'
      },
      {
        title: 'Lift Yourself'
      },
      {
        title: 'Gold Digger'
      }
    ];

    const wrapper = shallow(<Recordings artist={ 'Kanye West' } recordingArray={ fakeData }/>);
    expect(wrapper).toMatchSnapshot();
  });
});
