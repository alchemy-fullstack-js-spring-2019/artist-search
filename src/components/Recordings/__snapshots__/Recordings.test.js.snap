// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recordings component renders the Recordings 1`] = `
<ul>
  <li>
    <Recording
      artist="Kanye West"
      title="Stronger"
    />
  </li>
  <li>
    <Recording
      artist="Kanye West"
      title="Lift Yourself"
    />
  </li>
  <li>
    <Recording
      artist="Kanye West"
      title="Gold Digger"
    />
  </li>
</ul>
`;
